### PR TITLE
Approve squeak

### PR DESCRIPTION
**Pull request by gjcarew**
### User Story Implemented
An admin can approve or reject a squeak.

### Describe Features Added/Changed
See above. Also refactored the update method to use a helper method instead of a giant nested if block.

### Describe Tests Added/Changed
Added testing for approving and disapproving a squeak, and approving a squeak will reset reports to zero. 

### Checklist before requesting a review
- [x] Added features have thorough tests written for their functionality?
- [x] All Tests Passing?

### CODE REVIEW

**Review by AMSterling — COMMENTED** [review]
Be sure to update ReadME to reflect changes for FE

**Comment by gjcarew** [comment]
Thanks @AMSterling, just updated the README with a few more commits

File: app/graphql/mutations/update_squeak.rb [hunk 1]
```diff
@@ -3,22 +3,33 @@ class UpdateSqueak < Mutations::BaseMutation
...
-        end
-        squeak.save
+        check_arguments(squeak, report, nut, approved)
       end
       { squeak: squeak }
       rescue ActiveRecord::RecordNotFound => e
         GraphQL::ExecutionError.new(e)
     end
+
+    def check_arguments(squeak, report, nut, approved)
```
Comment (juliet-e) [inline app/graphql/mutations/update_squeak.rb:20]: Nice refactoring! This is much more readable than the longer conditional. 

This is a nit, but I wonder if `perform_update` or something could be a better name for this function. I was thinking it was going be parameter validation based on the name, but it is actually changing the state of the squeak

File: README.md [hunk 2]
```diff
@@ -461,7 +461,8 @@ Add a squeak.
    | Parameter | Description | Data type |
    | --------- | ----------- | --------- |
    | **content** | Squeak message content (required) | String        |
-   | **user_id** | Squeak message contant (required) | ID       |
+   | **user_id** | Squeak message content (required) | ID       |
+   | **approved** | Whether a squeak has been approved by a moderator | Boolean |
```
Comment (dara-rockwell) [inline README.md:465]: **praise:** Nice work in keeping the docs up to date!

File: README.md [hunk 3]
```diff
@@ -603,7 +605,37 @@ mutation {
 }
  ```
 
- ---
+ **Sample mutation (Approved)**
```
Comment (dara-rockwell) [inline README.md:608]: **praise:** Thorough!

File: app/graphql/mutations/update_squeak.rb [hunk 4]
```diff
@@ -3,22 +3,33 @@ class UpdateSqueak < Mutations::BaseMutation
     argument :id, ID, required: true
     argument :report, Boolean, required: false
     argument :nut, Boolean, required: false
+    argument :approved, Boolean, required: false
 
     field :squeak, Types::SqueakType, null: false
 
-    def resolve(id: , report: false, nut: false)
+    def resolve(id: , report: false, nut: false, approved: nil)
```
Comment (dara-rockwell) [inline app/graphql/mutations/update_squeak.rb:10]: **question:** What is resolve supposed to do?

File: app/graphql/mutations/update_squeak.rb (hunk 1 above)
Comment (dara-rockwell) [inline app/graphql/mutations/update_squeak.rb:20]: **suggestion:** Look into one of the many ruby validation libraries out there, or maybe consider using a PORO with `include ActiveModel::Model` to mix in validation logic.

File: spec/graphql/squeaks/update_squeak_spec.rb [hunk 5]
```diff
@@ -145,4 +145,87 @@
...
+                  metric
+                  probability
+                }
+              }
+            }
+          }
+        GQL
+      end
+
+      it 'The squeak status is changed to approved' do
```
Comment (dara-rockwell) [inline spec/graphql/squeaks/update_squeak_spec.rb:175]:
```suggestion
      it 'changes the squeak status to approved' do
```

File: spec/graphql/squeaks/update_squeak_spec.rb [hunk 6]
```diff
@@ -145,4 +145,87 @@
...
+        expect(@reported_squeak.approved).to be_nil
+
+        result = SqueakrBeSchema.execute(@query)
+        expect(result.dig("data")).to be_a(Hash)
+        expect(result.dig("data", "updateSqueak", "squeak")).to be_a(Hash)
+        squeak_return = result.dig("data", "updateSqueak", "squeak")
+        expect(squeak_return.dig("approved")).to be(true)
+
+        @reported_squeak.reload
+        expect(@reported_squeak.approved).to be(true)
```
Comment (dara-rockwell) [inline spec/graphql/squeaks/update_squeak_spec.rb:185]:
```suggestion
        result = SqueakrBeSchema.execute(@query)
        
        approved = result.dig("data", "updateSqueak", "squeak").dig("approved")
        
        @reported_squeak.reload
        expect(approved).to equal(@reported_squeak.approved)
```

**Review by dara-rockwell — APPROVED** [review]

**Review by Penitent0 — APPROVED** [review]
Great learning from Dara on this! My comment is not half-as-cool as the .gif!

**Pull request merged**